[Bug] Update cache fails if account PC contains non-AHV clusters (ESX)

**Describe the bug**
**init dsl** / **update cache** will fail if Nutanix PC account contains a mix of AHV and ESX clusters.

**Branch/Container**
Branch tag v3.5.0

**To Reproduce**
Steps to reproduce the behavior:
1. Configure a Nutanix Account connected to a PC with AHV and ESX clusters
2. Issue `calm update cache`
3. Observe error

**Expected behavior**
Successful completion of init dsl / update cache commands

**Findings/Troubleshooting Done**
In `dsl/db/table_config.py:AhvClusterCache.sync` it is attempting to create cache entries for AHV clusters. While it removes PC clusters from consideration, it assumes all clusters having the 'AOS' service as being AHV clusters when in fact other hypervisors may have the same attribute.  Additional filtering is necessary to remove non-AHV clusters from consideration.

https://github.com/nutanix/calm-dsl/blob/1aedf57e25cb3121c444d445216617e050cb0c97/calm/dsl/db/table_config.py#L421-L435

Changing line 422 as follows resolves the issue.

`if "AOS" not in service_list or name not in account_clusters_data_rev:`

**Screenshots**

```
$ calm update cache
[2022-08-05 20:26:17] [INFO] [calm.dsl.store.cache:157] Updating cache [2022-08-05 20:26:18] [WARNING] [calm.dsl.api.connection:311] Could not get PC Version..
[2022-08-05 20:26:26] [ERROR] [calm.dsl.db.table_config:453] PE Cluster UUID not supplied for AHV PE Cluster ABCD-EF-NTNX01
```

** System Configuration:**

- OS: Windows
- Python: 3.7.9
- Calm: 3.5.2
- PC: 2022.6

--- a/calm/dsl/cli/acps.py
+++ b/calm/dsl/cli/acps.py
@@ -302,7 +302,7 @@ def create_acp(role, project, acp_users, acp_groups, name):
         entity_filter_expression_list = ACP.ENTITY_FILTER_EXPRESSION_LIST.CONSUMER
 
     elif role == "Operator" and cluster_uuids:
-        entity_filter_expression_list = ACP.ENTITY_FILTER_EXPRESSION_LIST.CONSUMER
+        entity_filter_expression_list = ACP.ENTITY_FILTER_EXPRESSION_LIST.OPERATOR
 
     if cluster_uuids:
         entity_filter_expression_list.append(

--- a/calm/dsl/db/table_config.py
+++ b/calm/dsl/db/table_config.py
@@ -579,9 +579,8 @@ class AhvVpcsCache(CacheTableBase):
         cls.clear()
 
         client = get_api_client()
-        payload = {"length": 250}
+        payload = {"length": 250, "filter": "type==nutanix_pc"}
         account_name_uuid_map = client.account.get_name_uuid_map(payload)
-        account_uuid_type_map = client.account.get_uuid_type_map(payload)
         AhvVmProvider = cls.get_provider_plugin("AHV_VM")
         AhvObj = AhvVmProvider.get_api_obj()
 
@@ -590,14 +589,6 @@ class AhvVpcsCache(CacheTableBase):
         tunnel_entities = client.tunnel.list_all()
 
         for pc_acc_name, pc_acc_uuid in account_name_uuid_map.items():
-            acc_type = account_uuid_type_map.get(pc_acc_uuid)
-            if acc_type != "nutanix_pc":
-                LOG.debug(
-                    "Skipping account name: {} and uuid: {}".format(
-                        pc_acc_name, pc_acc_uuid
-                    )
-                )
-                continue
             try:
                 res = AhvObj.vpcs(account_uuid=pc_acc_uuid)
             except Exception:

--- a/calm/dsl/db/table_config.py
+++ b/calm/dsl/db/table_config.py
@@ -406,6 +406,8 @@ class AhvClustersCache(CacheTableBase):
                     )
                 )
                 continue
+
+            # TODO the order of cache sync is how their model is defined in table_config.py file
             account = AccountCache.get(uuid=pc_acc_uuid)
             account_clusters_data = json.loads(account.data).get("clusters", {})
             account_clusters_data_rev = {v: k for k, v in account_clusters_data.items()}
@@ -423,6 +425,15 @@ class AhvClustersCache(CacheTableBase):
                     LOG.debug(
                         "Cluster '{}' with UUID '{}' having function {} is not an AHV PE cluster".format(
                             name, uuid, service_list
+                        )
+                    )
+                    continue
+
+                # For esxi clusters, there will not be any pe account
+                if not account_clusters_data_rev.get(name, ""):
+                    LOG.debug(
+                        "Ignoring cluster '{}' with uuid '{}', as it doesn't have any pc account".format(
+                            name, uuid
                         )
                     )
                     continue
@@ -579,15 +590,13 @@ class AhvVpcsCache(CacheTableBase):
         cls.clear()
 
         client = get_api_client()
-        payload = {"length": 250, "filter": "type==nutanix_pc"}
+        payload = {"length": 250, "filter": "state==VERIFIED;type==nutanix_pc"}
         account_name_uuid_map = client.account.get_name_uuid_map(payload)
         AhvVmProvider = cls.get_provider_plugin("AHV_VM")
         AhvObj = AhvVmProvider.get_api_obj()
 
         # Get all Calm vpcs and Tunnels
         calm_vpc_entities = client.network_group.list_all()
-        tunnel_entities = client.tunnel.list_all()
-
         for pc_acc_name, pc_acc_uuid in account_name_uuid_map.items():
             try:
                 res = AhvObj.vpcs(account_uuid=pc_acc_uuid)
@@ -603,6 +612,7 @@ class AhvVpcsCache(CacheTableBase):
                 name = entity["status"]["name"]
                 uuid = entity["metadata"]["uuid"]
 
+                # TODO improve this, it shouldn't iterate over these entities every time
                 tunnel_reference = next(
                     (
                         calm_vpc["status"]["resources"].get("tunnel_reference", {})

